extensions/amp-access/0.1/test/test-amp-access.js

```diff
@@ -258,6 +258,44 @@ describe('AccessService authorization', () => {
       expect(elementOff).not.to.have.attribute('amp-access-off');
     });
   });
+
+  it('should resolve first-authorization promise after success', () => {
+    cidMock.expects('get')
+        .withExactArgs('amp-access', sinon.match(() => true))
+        .returns(Promise.resolve('reader1'))
+        .once();
+    xhrMock.expects('fetchJson')
+        .withExactArgs('https://acme.com/a?rid=reader1',
+            {credentials: 'include'})
+        .returns(Promise.resolve({access: true}))
+        .once();
+    return service.runAuthorization_().then(() => {
+      expect(service.firstAuthorizationPromise_).to.exist;
+      return service.firstAuthorizationPromise_;
+    });
+  });
+
+  it('should NOT resolve first-authorization promise after failure', () => {
+    cidMock.expects('get')
+        .withExactArgs('amp-access', sinon.match(() => true))
+        .returns(Promise.resolve('reader1'))
+        .once();
+    xhrMock.expects('fetchJson')
+        .withExactArgs('https://acme.com/a?rid=reader1',
+            {credentials: 'include'})
+        .returns(Promise.reject('intentional'))
+        .once();
+    return service.runAuthorization_().then(() => {
+      expect(service.firstAuthorizationPromise_).to.exist;
+      let resolved = false;
+      service.firstAuthorizationPromise_.then(() => {
+        resolved = true;
+      });
+      return Promise.resolve().then(() => {
+        expect(resolved).to.be.false;
+      });
```

Comment:
Ah, see https://github.com/ampproject/amphtml/pull/1350#discussion_r49245619. A rejected promise never fulfills, fixing the issue.